homeassistant/components/evse_wifi/sensor.py

```diff
@@ -0,0 +1,487 @@
+"""Platform for EVSE Sensor integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+import async_timeout
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    UnitOfElectricCurrent,
+    UnitOfEnergy,
+    UnitOfLength,
+    UnitOfPower,
+    UnitOfTime,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Add sensors for passed config_entry in HA."""
+    evse = hass.data[DOMAIN][config_entry.entry_id]
+    coordinator = EvseCoordinator(hass, evse)
+
+    new_devices = [
+        VehicleStateSensor(coordinator, evse),
+        VehicleStateTextSensor(coordinator, evse),
+    ]
+    new_devices.append(MaxCurrentSensor(coordinator, evse))
+    new_devices.append(ActualPowerSensor(coordinator, evse))
+    new_devices.append(DurationSensor(coordinator, evse))
+    new_devices.append(LastActionUserSensor(coordinator, evse))
+    new_devices.append(LastActionUidSensor(coordinator, evse))
+    new_devices.append(EnergySensor(coordinator, evse))
+    new_devices.append(MilageSensor(coordinator, evse))
+    new_devices.append(MeterReadingSensor(coordinator, evse))
+    new_devices.append(CurrentP1Sensor(coordinator, evse))
+    new_devices.append(CurrentP2Sensor(coordinator, evse))
+    new_devices.append(CurrentP3Sensor(coordinator, evse))
+    async_add_entities(new_devices)
+
+    await coordinator.async_config_entry_first_refresh()
+
+
+class EvseCoordinator(DataUpdateCoordinator):
+    """My custom coordinator."""
+
+    def __init__(self, hass, evse):
+        """Initialize my coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            # Name of the data. For logging purposes.
+            name="EVSE Data",
+            # Polling interval. Will only be polled if there are subscribers.
+            update_interval=timedelta(seconds=evse.interval),
+        )
+        self.evse = evse
+
+    async def _async_update_data(self):
+        """Fetch data from API endpoint."""
+        try:
+            # Note: asyncio.TimeoutError and aiohttp.ClientError are already
+            # handled by the data update coordinator.
+            async with async_timeout.timeout(10):
+                return await self.evse.get_parameters()
+        except Exception as exception:  # pylint: disable=broad-except
```

Comment:
Try not to use broad exception. Catch the actually exceptions you think may come up